README.md

```diff
@@ -1,5 +1,214 @@
 # react-carousel
 
-> Carousel in react :)
+> Feature-rich, react-way carousel component that does not suck 
 
-Under construction...
+[![CircleCI](https://circleci.com/gh/brainhubeu/react-carousel.svg?style=svg)](https://circleci.com/gh/brainhubeu/react-carousel)
+[![npm](https://img.shields.io/npm/v/@brainhubeu/react-carousel.svg)](https://www.npmjs.com/package/@brainhubeu/react-carousel)
+[![npm](https://img.shields.io/npm/l/@brainhubeu/react-carousel.svg)](https://www.npmjs.com/package/@brainhubeu/react-carousel)
+
+React carousel 
+
+## Why?
+There are some great carousels (like slick) that does not have real react implementations. This library provides you with carousel that is not merely wrapper for some jQuery solution, can be used as controlled or uncontrolled element (similar to [inputs](https://reactjs.org/docs/uncontrolled-components.html)), and has tons of useful features.
+
+## Installation
+`npm i @brainhubeu/react-carousel`
+
+## Usage
+By default the component does not need anything except children to render simple carousel.
+```javascript
+import Carousel from '@brainhubeu/react-carousel';
+
+...
+
+render() {
+  return (
+    <Carousel>
+      <img src={imageOne} />
+      <img src={imageTwo} />
+      <img src={imageThree} />
+    </Carousel>
+  );
+}
+```
+
+### Carousel as controlled element
+You can control which slides is being shown by providing Carousel with value and onChange props
+```javascript
+import Carousel from '@brainhubeu/react-carousel';
+
+...
+
+constructor(props) {
+  super(props);
+  this.state = {
+    value: 0,
+  };
+}
+
+// we check if we got event from input (and it has target property) or just value from Carousel 
+onChange = e => this.setState({ value: e.target ? e.target.value : e });
+render() {
+  return (
+    <div>
+      <input value={this.state.value} onChange={this.onChange} type="number" />
+      <Carousel
+        value={this.state.value}
+        onChange={this.onChange}
+      >
+        <img src={imageOne} />
+        <img src={imageTwo} />
+        <img src={imageThree} />
+      </Carousel>
+    </div>
+  );
+}
+```
+
+Where:
+* `value` is current slide index (zero based, in the example above imageOne has index 0, imageTwo has index 1 and so on)
```

Comment:
`value is the current slide's index` or `value is the index of the current slide`